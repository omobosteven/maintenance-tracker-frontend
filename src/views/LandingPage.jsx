import React from 'react';
import { Link } from 'react-router-dom';
import routes from '../constants/routes';

const LandingPage = () => (
  <section className="row main-content">
    <h1 className="main-heading">Maintenance Tracker</h1>
    <p className="main-text">Lorem Ipsum cool text right here...</p>
    <div className="main-buttons">
      <Link to={routes.SIGN_UP} className="btn main-btn">Signup</Link>
      <Link to={routes.SIGN_IN} className="btn main-btn">Login</Link>
    </div>
  </section>
);

export default LandingPage;
