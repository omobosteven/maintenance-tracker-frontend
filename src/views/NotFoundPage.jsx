import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
  <section className="section-requests container">
    <div className="row">
      <h1 className="section-heading">
        Oops!!!, You might want to
        <Link to="/" className="not-found"> Click me</Link>
      </h1>
    </div>
    404!!!
  </section>
);

export default NotFoundPage;
