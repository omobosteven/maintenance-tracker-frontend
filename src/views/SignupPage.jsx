import React from 'react';

const SignupPage = () => (
  <div>
    This is from the Signup component
  </div>
);
export default SignupPage;
