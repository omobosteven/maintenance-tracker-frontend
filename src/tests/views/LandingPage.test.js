import React from 'react';
import { shallow } from 'enzyme';
import { LandingPage } from '../../views/LandingPage';

let wrapper;

describe('LandingPage component', () => {
  beforeEach(() => {
    const props = {
      auth: false
    };

    wrapper = shallow(<LandingPage {...props} />);
  });

  test('should render a snapshot of the LandingPage component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should redirect user if logged in', () => {
    const props = {
      auth: true
    };

    wrapper = shallow(<LandingPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
