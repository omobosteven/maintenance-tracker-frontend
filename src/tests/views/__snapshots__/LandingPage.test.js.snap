// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component should redirect user if logged in 1`] = `
<Redirect
  push={false}
  to="/users/requests"
/>
`;

exports[`LandingPage component should render a snapshot of the LandingPage component 1`] = `
<section
  className="row main-content"
>
  <h1
    className="main-heading"
  >
    Maintenance Tracker
  </h1>
  <p
    className="main-text"
  >
    Lorem Ipsum cool text right here...
  </p>
  <div
    className="main-buttons"
  >
    <Link
      className="btn main-btn"
      replace={false}
      to="/auth/signup"
    >
      Signup
    </Link>
    <Link
      className="btn main-btn"
      replace={false}
      to="/auth/login"
    >
      Login
    </Link>
  </div>
</section>
`;
